File: Coding-Challenges/ArrayWaterArea/ArrayWaterContainer.py
def ArrayWaterContainer(height: list[int]):
    leftIndex = 0
    rightIndex = len(height) - 1
    max = 0
    while (leftIndex != rightIndex):
        localHeight = height[leftIndex] if height[leftIndex] < height[rightIndex] else height[rightIndex]
        max = localHeight * (rightIndex - leftIndex) if localHeight * (rightIndex - leftIndex) > max else max
        if (height[leftIndex] > height[rightIndex]):
            rightIndex -= 1
        else:
            leftIndex += 1
    return max

print(ArrayWaterContainer([1,8,6,2,5,4,8,3,7]))

File: Coding-Challenges/ArrayWaterArea/ArrayWaterContainer.java
public class ArrayWaterContainer{
    public static void main(String args[]){
        ArrayWaterContainer tester = new ArrayWaterContainer();
        int[] heights = {1, 8, 6, 2, 5, 4, 8, 3, 7};
        System.out.println(tester.findMaxArea(heights));
    }

    public int findMaxArea(int[] height){
        int leftIndex = 0;
        int rightIndex = height.length - 1;
        int max = 0;
        while (leftIndex != rightIndex){
            int localHeight = height[leftIndex] < height[rightIndex] ? height[leftIndex] : height[rightIndex];
            max = localHeight * (rightIndex - leftIndex) > max ? localHeight * (rightIndex - leftIndex) : max;
            if (height[leftIndex] < height[rightIndex])
                leftIndex++;
            else
                rightIndex--;
        }
        return max;
    }
}